compiler/rustc_middle/src/ty/trait_def.rs

```diff
@@ -45,6 +45,12 @@ pub struct TraitDef {
     pub must_implement_one_of: Option<Box<[Ident]>>,
 }
 
+#[derive(HashStable, Encodable, Decodable)]
+pub struct TraitAliasDef {
+    pub def_id: DefId,
+    pub explicit_bounds: Box<[DefId]>,
```

Comment:
Is it sensible to directly store the bounds' `DefId`s here, or would some intermediate type be preferable?